assessments/projects/fluentd/self-assessment.md

```diff
@@ -0,0 +1,223 @@
+- **Forward**:
+  The input and output plugin that listens to the TCP socket to receive the event stream, which allows event log retrieval from other Fluentd instances. Safely transmitting data between Fluentd instances with the help of TCP.
+- **Memory**:
+  The buffer plugin that uses memory to store buffer chunks. When Fluentd is shut down, buffered logs that cannot be written quickly are deleted to avoid incomplete data written into the memory.
+
+
+### Security Relevant:
+1. Pre-installation
+   It is recommended to
+  * Set up an **NTP daemon** for accurate current timestamp which is essential for product-grade logging services.
+  * Set up [AWS-hosted NTP server](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-time.html) for AWS users.
+  * Increase the **Maximum Number of File Descriptors** to guarantee sufficient file descriptors for Fluentd.
+  * Optimize the **Network Kernal Parameters** to secure smooth network connection for Fluentd.
+  * Use **sticky bit symlink/hardlink protection** to ensure link security because Fluentd sometimes uses predictable paths for dumping, writing files, and so on.
+
+2. Logging
+   Fluentd has two logging layers: global and per plugin, combined they grant the ability to track all events that occurred, including security-related ones. This is important for monitoring and analysis, providing observability regarding security issues.
+
+3. Memory Buffering
+   When the destination Fluentd instance dies, certain logger implementations will use extra memory to hold the incoming logs that would be immediately sent again once Fluentd comes back, ensuring no critical security events and activities discarded that might affect the overall safety.
+   
+## Project compliance
+* The Fluentd project has earned the badge of [Open Source Security Foundation (OpenSSF)](https://www.bestpractices.dev/en/projects/1189#reporting), illustrating its qualification regarding security, quality, and various other fields in detail.
+
+## Secure development practices
+
+### Development Pipeline
+All code is maintained in a public repository on GitHub, and all changes need to be reviewed by the maintainers. See [Fluentd workflow](https://github.com/fluent/fluentd/blob/master/GithubWorkflow.md) for the detailed pipeline for contributors.
+- All source code are publicly available on GitHub.
+- Manually test the code through `bundle exec rake test` before PR.
+- All changes are made to a forked repository under the branch named **new-feature**.
+- Commit the final changes to the original repository through PR.
+- All the commits requires to be signed by the contributor under the regulation of [Developer Certificate of Origin](https://probot.github.io/apps/dco/).
+- Once the PR is created, it will be assigned to one or more reviewers.
+- Merging automatically occurs after the PR is approved by the reviewer and contributor.
```

Comment:
Is any form of secret scanning, SAST or SCA done in the CI pipeline? This would be the place to document it.

Reply:
No there is not. Confirmed with a maintainer.